src/engraving/compat/midi/compatmidirender.cpp

```diff
@@ -404,15 +404,13 @@ void CompatMidiRender::renderTremolo(Chord* chord, std::vector<NoteEventList>& e
             events->clear();
         }
     } else if (chord->tremoloChordType() == TremoloChordType::TremoloSingle) {
-        int t = Constants::DIVISION / (1 << (tremolo.lines() + chord->durationType().hooks()));
-        if (t == 0) {       // avoid crash on very short tremolo
-            t = 1;
-        }
+        double tpoch = muse::RealIsNull(tremoloPartOfChord) ? 1.0 : tremoloPartOfChord;
 
-        int tremoloEventsSize = chord->ticks().ticks() / t * tremoloPartOfChord;
+        int tremoloEventsSize = static_cast<int>(static_cast<double>(chord->ticks().ticks())
+                                                 / (static_cast<double>(t) * tpoch));
```

Comment:
I believe the `double` casts don't make a difference here: when `(static_cast<double>(t) * tpoch) > chord->ticks().ticks()`, then the result of the division would be between 0.0 and 1.0, so casting that back to `int` will still result in 0. 

But you are changing the meaning of the code. Essentially, this is what's happening:
```diff
-(chord->ticks().ticks() / t) * tremoloPartOfChord
+chord->ticks().ticks() / (t * tremoloPartOfChord)
```
I think that's incorrect: if the whole chord were a tremolo, then `(chord->ticks().ticks() / t)` would be the correct number of events; and if only part of the chord is a tremolo (i.e. `tremoloPartOfChord < 1`) then `(chord->ticks().ticks() / t) * tremoloPartOfChord` would be the correct number.

It looks like the calculation of `tremoloEventsSize` can stay as it is (even extra casts to double would not have any effect); all that's needed would be to check that `tremoloEventsSize` is not zero. If it is zero, then just do an early return, since the `for` loop below would not be run anyway in that case. 

Apart from that, the precision of the calculations could of course be improved, by utilising `Fraction` and/or `double` everywhere, i.e. in `t`, `tremoloEventsSize`, `tremoloTime`, `tremoloEventStep`; and cast to `int` only at the very end. (Note that that ensures so that `t != 0`, since that would only happen as a result of rounding errors.)

Reply:
Ok
Initially I had a problem in another branch and this branch just doesn't have this check which you've mentioned.
```
    if (muse::RealIsNull(tremoloPartOfChord)) {
        return;
    }
```

And this more than enough.
Thank you for your help and time, @cbjeukendrup 

Closing this PR.